Automattic/woocommerce-payments
# [Settings] Add shortened customer bank statement

#### Changes proposed in this Pull Request

This is the first part of two PRs to get the statement descriptor info sent in the payment intents. That information seems to be sent over inconsistently across the many ways we use to create/update the intent.

In this part, the shortened descriptor is being added to the settings. The user should be able to provide a shortened version of the descriptor that will be concatenated with the order number. The shortened version is meant to be used for card transactions only.

See the design below. The focus of this PR is the functionality, hence the preview is not included – it can be addressed in another PR later on.

![image](https://user-images.githubusercontent.com/10233985/149021002-3137398a-2475-4519-a086-35d5fd3d587d.png)









#### Testing instructions





1. Go to the WCPay [settings](http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments).
2. You should see a new unticked checkbox `Add customer order number to the bank statement` below the regular bank statement.
3. Tick it and the `Shortened customer bank statement` field should appear.
4. Type in something and save.
5. Play with some values in the `Shortened customer bank statement` field. Validation should work the same way as for the regular bank statement, except for the field max length.
6. Untick `Add customer order number to the bank statement` and `Shortened customer bank statement` field should disappear.
7. `Full bank statement` should work as usual.

-------------------


**Post merge**
